src/diffusers/onnx_utils.py

```diff
@@ -0,0 +1,264 @@
+# coding=utf-8
+# Copyright 2022 The HuggingFace Inc. team.
+# Copyright (c) 2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import os
+import shutil
+import subprocess
+from pathlib import Path
+from typing import Optional, Union
+
+import numpy as np
+
+import onnxruntime as ort
+import requests
+from huggingface_hub import HfApi, HfFolder, hf_hub_download
+
+from .utils import logging
+
+
+ONNX_WEIGHTS_NAME = "model.onnx"
+
+
+logger = logging.get_logger(__name__)
+
+
+class OnnxModel:
+    base_model_prefix = "onnx_model"
+
+    def __init__(self, model=None, **kwargs):
+        self.model = model
+        self.model_save_dir = kwargs.get("model_save_dir", None)
+        self.latest_model_name = kwargs.get("latest_model_name", "model.onnx")
```

Comment:
This is mostly copied from [`optimum.onnxruntime.ORTModel`](https://github.com/huggingface/optimum/blob/f5a741330bef093b6ffe0023455dff722e4515e8/optimum/onnxruntime/modeling_ort.py#L79) 

The copy was necessary to remove the AutoModel and AutoConfig-related code (the optimum model was intended as a wrapper for transformers models only)

Reply:
Okey for me! Think we could always adapt later on

Reply:
good for me as well